doc/design/refactor_discuess/readme.md

```diff
@@ -0,0 +1,273 @@
+# Paddle 计算重构设计文档
+
+整体设计如下图所示:
+
+![alt](./pics/whole_archi.png)
+
+主要的概念包括:
+
+* Workspace: 全部的神经网络层的参数和输入，均由一个全局对象所管理。内存/显存的申请，释放，resize都使用该对象管理。
+	* 内存申请后的类型是`TensorBuffer`，该类型是一个`void*`指针，size
+* Tensor是计算Kernel函数的参数格式，他包括`TensorBuffer`和`TensorAttr`。其中，`TensorAttr`中包括了Tensor的`Shape`，是否需要Gradient，设备信息等。
```

Comment:
Obviously, a "tensor" is a high-dimensional matrix, and cannot "include gradient", because a gradient is another tensor.